test/fuzz/common.proto

```diff
@@ -15,9 +16,29 @@ message Headers {
   repeated envoy.config.core.v3.HeaderValue headers = 1;
 }
 
+message HttpBody {
+  // The bytes that will be used as the request body.
+  repeated string data = 1 [(validate.rules).repeated .min_items = 1];
+}
+
+// HttpBody cannot efficiently create serialized protos.
+// Use ProtoBody instead to test grpc data.
+message ProtoBody {
+  // The proto message that will be serialized and used as the request body.
+  google.protobuf.Any message = 1 [(validate.rules).any.required = true];
+
+  // The size (in bytes) of each buffer when forming the requests.
+  uint64 chuck_size = 2 [(validate.rules).uint64.gt = 0];
+}
+
 message HttpData {
   Headers headers = 1;
-  repeated string data = 2;
+
+  oneof body {
```

Comment:
@vitalybuka Quick question about LPM: If we break wire compatibility in our protos, we'd have to wait for the corpus to re-generate from scratch, right? Do you have any suggestions for a better way to add this?
If we didn't want to break compatibility could define add the new message as is, and use a bool to switch instead of the oneof.

Reply:
In general yes, some conversion is needed.

However, for for example we maybe can add some workaround.
That's how code parsed
https://github.com/google/libprotobuf-mutator/blob/f48c32bd48bfbef9b7e7989c542ffbdc1f360605/src/binary_format.cc#L27

so when field is remove it's likely entire message will be cleared. but I am not 100% sure what is behavior of ParsePartialFromString here

we don't want invalid messages, e.g. without required fileds, to be used in fuzzer.
maybe instead of ->Clear() we should keep whatever was parsed and instead created some default values for missing required fields.

Reply:
Actually maybe this is fine? With the original field removed, `ParseFromString` or `ParsePartialFromString` would treat that as an unknown field. We know protos allow unknown fields, so that parse shouldn't fail.

So essentially the corpus will still be valid, just the HTTP payloads will all be empty. But letting the corpus generate that doesn't sound too bad (anyways, most filters operate on headers).

Reply:
Actually I think we're mixing up to things here: the binary serialized format and the text format.

Our corpus consists of text format protos. From my understanding, unknown fields when parsing text format is not allowed.

libprotobufmutator uses `ParseFromString` for parsing text format. Based on  protocolbuffers/protobuf#5465, it's not clear if this allows unknown fields.

https://github.com/google/libprotobuf-mutator/blob/cdb7deb9f17379545873c256b139537ca40bcc5e/src/text_format.cc#L33

I believe using `Merge` would solve this issue: https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.text_format#TextFormat.Merge

> Like Parse(), but the data is merged into the given message, as if using Message::MergeFrom().

Reply:
To wrap this up, the resolution is adding unknown field support with https://github.com/google/libprotobuf-mutator/pull/170 so the fuzz target will still parse the existing corpus with empty `http_body` and `proto_body`. Is that right? If so, you'll need to bump the libprotobuf-mutator versions when the PR is merged.

Reply:
Yup, that is the plan. Updating now.